mlflow/tracking/_model_registry/fluent.py

```diff
@@ -428,7 +429,14 @@ def load_prompt(name_or_uri: str, version: Optional[int] = None) -> Prompt:
         prompt = mlflow.load_prompt("prompts:/my_prompt@production")
 
     """
-    return MlflowClient().load_prompt(name_or_uri=name_or_uri, version=version)
+    client = MlflowClient()
+    prompt = client.load_prompt(name_or_uri=name_or_uri, version=version)
+
+    # If there is an active MLflow run, associate the prompt with the run
+    if run := active_run():
+        client.log_prompt(run.info.run_id, f"prompts:/{prompt.name}/{prompt.version}")
+
+    return prompt
```

Comment:
This change also makes me wonder if we should make run_id optional in log_prompt API, and use current active run if run_id is None

Reply:
Oh never mind, run_id is optional in create_model_version